typescript/agentkit/src/wallet-providers/nearKeypairWalletProvider.ts

```diff
@@ -0,0 +1,264 @@
+   *
+   * @returns The public key of the keypair.
+   */
+  getPublicKey(): string {
+    return this.keypair.getPublicKey().toString();
+  }
+
+  /**
+   * Get the address of the wallet provider.
+   *
+   * @returns The address of the wallet provider.
+   */
+  getAddress(): string {
+    return this.accountId;
+  }
+
+  /**
+   * Get the network of the wallet provider.
+   *
+   * @returns The network of the wallet provider.
+   */
+  getNetwork(): Network {
+    return NEAR_NETWORKS[this.network];
+  }
+
+  /**
+   * Get the name of the wallet provider.
+   *
+   * @returns The name of the wallet provider.
+   */
+  getName(): string {
+    return "near_keypair_wallet_provider";
+  }
+
+  /**
+   * Get the connection of the wallet provider.
+   *
+   * @returns The connection of the wallet
+   */
+  async getConnection(): Promise<Connection> {
+    await this.connectIfNeeded();
+    return this.connection;
+  }
+
+  /**
+   * Get the NEAR account of the wallet provider.
+   *
+   * @returns The NEAR account of the wallet provider.
+   */
+  async getAccount(): Promise<Account> {
+    await this.connectIfNeeded();
+    return this.account!;
+  }
+
+  /**
+   * Get the balance of the native asset of the network.
+   *
+   * @returns The balance of the native asset of the network.
+   */
+  async getBalance(): Promise<bigint> {
+    await this.connectIfNeeded();
+    const balance = await this.account.getAccountBalance();
+    return BigInt(balance.available);
+  }
+
+  /**
+   * Transfer the native asset of the network.
+   *
+   * @param to - The destination address.
+   * @param value - The amount to transfer in whole units (e.g. ETH)
+   * @returns The transaction hash
+   */
+  async nativeTransfer(to: string, value: string): Promise<string> {
+    await this.connectIfNeeded();
+    const amountYocto = nearUtils.format.parseNearAmount(value);
+    if (!amountYocto) {
+      throw new Error("Error converting amount to yoctonear");
+    }
+    const action = actionCreators.transfer(BigInt(amountYocto));
+    const outcome = await this.account.signAndSendTransaction({
+      receiverId: to,
+      actions: [action],
+    });
+    return outcome.transaction.hash;
+  }
+
+  /**
+   * Sign and send a transaction.
+   *
+   * @param args - The transaction sender parameters.
+   * @param options - The send transaction options.
+   *
+   * @returns The final execution outcome
+   */
+  async signAndSendTransaction(
+    args: TransactionSenderParams,
+    options: SendTransactionOptions = DEFAULT_OPTIONS,
+  ): Promise<FinalExecutionOutcome> {
+    await this.connectIfNeeded();
+    const accountId = this.accountId;
+
+    const { signer } = this.near.connection;
+    const connection = this.near.connection;
+
+    const publicKey = await signer.getPublicKey(accountId, connection.networkId);
+
+    const accessKey = (await connection.provider.query(
+      `access_key/${accountId}/${publicKey.toString()}`,
+      "",
+    )) as unknown as {
```

Comment:
nit: If it has to be `as unknown`, that's fine. However, please try and see if there's a better typed way to accomplish this.

I find quite often this is a sign of the typings struggling with wrapping/unwrapping promises. So please try this first:
```ts
const accessKey = (await connection.provider.query(
      `access_key/${accountId}/${publicKey.toString()}`,
      "",
    )) as Promise<{
      block_hash: string;
      block_height: number;
      nonce: number;
      permission: string;
}>
```

Reply:
applied change